File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/achievement/AchievementFinishedMessage.scala





















// Generated on 05/03/2013 10:34:32
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait AchievementFinishedMessage extends MessageDeserializer {
  val messageId = 6208

  protected trait Message extends BaseMessage with Serializable {
    val messageId = AchievementFinishedMessage.messageId

    val id: Short
    val finishedlevel: Short

    def serialize(buf: Buffer) {
      buf.writeShort(id)
      buf.writeShort(finishedlevel)
    }

  }

  protected def create(id: Short, finishedlevel: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val id = buf.readShort()
    val finishedlevel = buf.readShort()

    create(id, finishedlevel)
  }

}

object AchievementFinishedMessage extends AchievementFinishedMessage {
  case class MessageImpl(id: Short, finishedlevel: Short) extends Message
  type Target = MessageImpl

  override def create(id: Short, finishedlevel: Short) = MessageImpl(id, finishedlevel)

  def apply(id: Short, finishedlevel: Short) = create(id, finishedlevel)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/AchievementFinishedMessage.java


// Generated on 05/08/2013 19:37:38
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class AchievementFinishedMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6208;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short id;
    public short finishedlevel;
    
    public AchievementFinishedMessage() { }
    
    public AchievementFinishedMessage(short id, short finishedlevel) {
        this.id = id;
        this.finishedlevel = finishedlevel;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(id);
        buf.writeShort(finishedlevel);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        id = buf.readShort();
        if (id < 0)
            throw new RuntimeException("Forbidden value on id = " + id + ", it doesn't respect the following condition : id < 0");
        finishedlevel = buf.readShort();
        if (finishedlevel < 0 || finishedlevel > 200)
            throw new RuntimeException("Forbidden value on finishedlevel = " + finishedlevel + ", it doesn't respect the following condition : finishedlevel < 0 || finishedlevel > 200");
    }
    
}
